src/dbus-function-ifaces/gadgetd-serial-function-iface.c

```diff
@@ -120,15 +122,16 @@ function_serial_attrs_get_property(GObject    *object,
 		return;
 	}
 
-	usbg_ret = usbg_get_function_attrs(f, &f_attrs);
+	f_serial = usbg_to_serial_function(f);
+	usbg_ret = usbg_f_serial_get_port_num(f_serial, &port_num);
 	if (usbg_ret != USBG_SUCCESS) {
-		ERROR("Cant get function attributes");
+		ERROR("Cant get serial port number");
 		return;
 	}
 
 	switch(property_id) {
 	case PROP_SERIAL_PORTNUM:
-		g_value_set_int(value, f_attrs.serial.port_num);
+		g_value_set_int(value, port_num);
```

Comment:
to be honest, as we use a specified getter I would move it's call inside this case